File: core/src/main/java/com/emarsys/core/response/AbstractResponseHandler.java
package com.emarsys.core.response;

import com.emarsys.core.util.Assert;

public abstract class AbstractResponseHandler {

    public void processResponse(ResponseModel responseModel) {
        Assert.notNull(responseModel, "ResponseModel must not be null");
        if (shouldHandleResponse(responseModel)) {
            handleResponse(responseModel);
        }
    }

    protected abstract boolean shouldHandleResponse(ResponseModel responseModel);

    protected abstract void handleResponse(ResponseModel responseModel);

}

File: core/src/main/java/com/emarsys/core/response/AbstractResponseHandler.kt
package com.emarsys.core.response

abstract class AbstractResponseHandler {

    fun processResponse(responseModel: ResponseModel) {
        if (shouldHandleResponse(responseModel)) {
            handleResponse(responseModel)
        }
    }

    abstract fun shouldHandleResponse(responseModel: ResponseModel): Boolean

    abstract fun handleResponse(responseModel: ResponseModel)

}
